Feature/routing

PReichetanz opened the pull request:
Implement Routing to App

- Add routes to App.tsx
- Add deleteEvaluation function
- Add EvaluationCard component (mostly copied from Card component, but less content)
- Refactor Card component
- Uplift BrowserRouter to main.tsx (in order to use useNavigate hook in App.tsx)
- Create ClassOverview.tsx and PupilOverview.tsx and move existing content display there)

PReichetanz committed:
Add basic Routing

- Move content from App.tsx to ClassOverview.tsx
- Add Routing to App

PReichetanz committed:
Add handleClick to Card

- handleClick: navigate to PupilPage

PReichetanz committed:
Add useNavigate to ClassOverview

PReichetanz committed:
Refactor Card: handleClick not optional anymore

PReichetanz committed:
Move BrowserRouter to main.tsx

PReichetanz committed:
Refactor App

- Lift state up (again) from ClassOverview = reverse previous changes
- Add deleteEvaluation function
- Add Route to Class- and PupilOverview
- Add PupilHandler function for PupilOverview

PReichetanz committed:
Add component EvaluationCard

- only slightly different to Card component

PReichetanz committed:
Refactor ClassOverview (lifting state up)

- Lift state up to App.tsx
- Add props type and adjust function

PReichetanz committed:
Add PupilOverview page

PReichetanz committed:
Update Card story with new handleClick function

marwinburesch commented on src/app/components/Card/Card.tsx:
```diff
@@ -8,12 +8,17 @@ type CardProps = {
   const { name, evaluations } = pupil;
   return (
-    <Container>
+    <Container onClick={handleClick}>
```
This is also triggered when `RoundActionButton` is clicked. I recommend using a link as a wrapper for this container instead of the onclick prop

marwinburesch commented on src/app/components/EvaluationCard/EvaluationCard.tsx:
```diff
@@ -0,0 +1,54 @@
+  header,
+  name,
+  evaluation,
+  deleteCard,
```
I think it would make more sense to call this prop `onDeleteClick` or something that describes the event basically. You're not actually deleting a `Card` just some data that previously led to the `Card` being generated. Also, `deleteCard` describes what will happen, but from the Card's point of view, the actual functionality will be assigned later.

marwinburesch reviewed (commented):
Good stuff Paul!
Only the `App.tsx`  structure needs some shuffling around to make it more streamlined, as we discussed earlier. Pages shouldn't have or need props. You have all the tools (`localStorage` hook and `useParams`) to access central data without having a state in your `App.tsx`
- Move logic in this PR
- add custom hook for pupils in the next PR :)

PReichetanz committed:
Make changes according to review

PReichetanz committed:
Refactor App.tsx

- Move all content to pages
- Only routing left

PReichetanz committed:
Add Link to Card

- Remove Link from ClassOverview when Card is called
- Reason: DeleteButton was a Link as well

PReichetanz committed:
Refactor onDeleteClick to handle DeleteEvaluation

PReichetanz committed:
Add content from App to pages
